docs/HowToPortTizenRTOnWiFiChipset.md

```diff
@@ -0,0 +1,141 @@
+
+There are two cases to consider towards porting TizenRT on a new WiFi chipset. The specific case depends on the vendor's decision to either use their proprietary supplicant codebase, or
+to directly use TizenRT's WiFi solution.
+
+Below are mentioned the two cases in detail. Additionally, instructions are provided to incorporate the WiFi chipset driver in TizenRT's code tree. Specifically, details are provided
+to interface driver to network stack, and to get/set WiFi configuration from a higher abstraction layer.
+ 
+## How to add external WiFi library to TizenRT
+
+External vendors may choose to build TizenRT application that includes their compiled WiFi libary, rather than opting for TizenRT's *wpa_supplicant*.
+To support external and independently compiled WiFi supplicant libraries, they must be merged with *libexternal.a* static libary that resides in the *external* folder of TizenRT.
+For details on how to merge libraries, please refer to [How to add Static Library](HowToAddStaticLibrary.md).
+
+
+
+## How to use WPA_SUPPLICANT
+If the developer wishes to use TizenRT's *wpa_supplicant* for WiFi-based applications, then following changes need to be performed at *[wpa_suppliant](#supplicant)*.
+
+### Supplicant
+
+Changes in *external* directory for a new WiFi board to include new chipset driver.
+
+1. Make sure you have defined *CONFIG_DRIVER_\<DRIVER_NAME\>* in *external/wpa_supplicant/Kconfig*. Refer to the example below for *CONFIG_DRIVER_T20*, which
+is subject to the condition that *CONFIG_SCSC_WLAN* is enabled already.
+```
+config DRIVER_T20
+	bool "Driver T20 for SLSI WiFi"
+	default y
+	depends on SCSC_WLAN
+```
+
+2. Add new chipset driver under *external/wpa_supplicant/src/drivers/*, and name it as
+*driver_\<driver_name\>.c*. Select this file for build in *external/wpa_supplicant/src/drivers/Make.defs*, depending on the Kconfig that is defined for your WiFi driver.
+An example is shown for LSI WiFi:
+
+```
+ifeq ($(CONFIG_DRIVER_T20), y),
+CSCRS += driver_t20.c
+endif
+```
+
+3. Inside *driver_\<driver_name\>.c*, declare a driver structure variable named *wpa_driver_\<driver_name\>_ops*.
+*wpa_driver_\<driver_name\>_ops* is a structure of type *wpa_driver_ops* that is declared in *external/wpa_supplicant/src/drivers/driver.h*.
+The *wpa_driver_ops* structure contains function pointers to the specific WiFi driver that the supplicant links to.
+As an example, refer to the *wpa_driver_t20_ops* structure declared in *external/wpa_supplicant/src/drivers/driver_t20.c* for the LSI WiFi chipset.
+```
+const struct wpa_driver_ops wpa_driver_t20_ops = {
+	.name = "slsi_t20",
+	.desc = "SLSI T20 Driver",
+	.init2 = slsi_t20_init,
+	.deinit = slsi_t20_deinit,
+	.get_mac_addr = slsi_get_mac_addr,
+	.get_capa = slsi_t20_get_capa,
+	.scan2 = slsi_hw_scan,…
+} 
+```
+
+4. Add the driver structure variable created in step 2 to the *wpa_drivers* list in *external/wpa_supplicant/src/drivers/drivers.c*, as shown below:
```

Comment:
Step 3 is correct. Step 2 is adding new file but you want to mention adding new structure.